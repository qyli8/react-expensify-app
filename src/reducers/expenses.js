//Expenses Reducer
const expensesReducerDefaultState=[]

const expensesReducer = (state=expensesReducerDefaultState, action) =>{
  switch(action.type){
    case 'ADD_EXPENSE':
    // spread operator allow concatination and create new array
     return [...state, action.expense]
    case 'REMOVE_EXPENSE':
      return state.filter(i=>i.id.localeCompare(action.id)!=0)
    case 'EDIT_EXPENSE':
      return state.map( (expense)=>{
        if(expense.id === action.id){
          return ({
            ... expense,
            ...action.updates
          })
        }
        else{
          return expense
        }
      })
    default:
    return state
  }
}

export default expensesReducer;