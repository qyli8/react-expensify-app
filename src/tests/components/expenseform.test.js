import React from 'react'
import {shallow} from 'enzyme';
import moment from 'moment'
import ExpenseForm from '../../components/ExpressForm'
import expenses from '../fixture/expenses'

test('should return expense form',()=>{
  const wrapper = shallow(<ExpenseForm/>)
  expect(wrapper).toMatchSnapshot()

})
test('should return expense form with expese data',()=>{
  const wrapper = shallow(<ExpenseForm expense={expenses[1]}/>)
  expect(wrapper).toMatchSnapshot()

})

test('should render error for invalid form submission',()=>{
  const wrapper = shallow(<ExpenseForm />)
  expect(wrapper).toMatchSnapshot()
  wrapper.find('form').simulate('submit', {
    preventDefault: ()=>{}
  });
  expect(wrapper.state('error').length).toBeGreaterThan(0)
  expect(wrapper).toMatchSnapshot()
})

test('should set description on input change', ()=>{
  const value="New Description"
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('input').at(0).simulate('change', {
    target:{value}
  })
  expect(wrapper.state('description')).toBe(value)
  expect(wrapper).toMatchSnapshot()
})

test('should set note on text change', ()=>{
  const value="some notes"
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('textarea').at(0).simulate('change', {
    target:{value}
  })
  expect(wrapper.state('note')).toBe(value)
  expect(wrapper).toMatchSnapshot()
})

test('should set amount if valid',()=>{
  const value = '23.50'
  const wrapper = shallow(<ExpenseForm />)
  wrapper.find('input').at(1).simulate('change',{
    target: {value}
  })
  expect(wrapper.state('amount')).toBe(value)
  expect(wrapper).toMatchSnapshot()
})

test('should not set amount if invalid input',()=>{
  const value ='12.222'
  const wrapper = shallow(<ExpenseForm />)
  wrapper.find('input').at(1).simulate('change',{
    target: {value}
  })
  expect(wrapper.state('amount')).toBe('')
  expect(wrapper).toMatchSnapshot()
  
})

test('should call onsubmit prop for valid form submission',()=>{
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<ExpenseForm expense={expenses[0]} onSubmit={onSubmitSpy} />);
  wrapper.find('form').simulate('submit', {
    preventDefault:()=>{}
  });
  expect(wrapper.state('error')).toBe('')
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
      "description": expenses[0].description, 
      "amount": parseFloat(expenses[0].amount), 
      "note": expenses[0].note,
      "createdAt": expenses[0].createdAt
    })
})

test('should set new date on date change', ()=>{
  const now = moment();
  const wrapper = shallow(<ExpenseForm/>)
  wrapper.find('SingleDatePicker').prop('onDateChange')(now);
  expect(wrapper.state('createdAt')).toEqual(now);
})

test('should set calendar focus on change',()=>{
  const focused = true
  const wrapper = shallow(<ExpenseForm/>)
  wrapper.find('SingleDatePicker').prop('onFocusChange')({focused})
  expect(wrapper.state('calendarFocused')).toBe(focused)
})
