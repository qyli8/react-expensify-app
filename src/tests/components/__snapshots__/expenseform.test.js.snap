// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not set amount if invalid input 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__itemLong"
      >
        <input
          autoFocus={true}
          className="input__long"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Amount"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <textarea
      className="textarea"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button
      className="button-sytle__edit"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`should render error for invalid form submission 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <CustomInput
      formControlProps={
        Object {
          "fullWidth": true,
        }
      }
      id="float"
      inputProps={
        Object {
          "placeholder": "Description",
          "startAdornment": <Unknown
            position="start"
        >
            <Unknown
                size="2.5rem"
            />
        </Unknown>,
        }
      }
      onChange={[Function]}
      value=""
    />
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <CustomInput
          formControlProps={
            Object {
              "fullWidth": true,
            }
          }
          id="float"
          inputProps={
            Object {
              "placeholder": "Cost",
              "startAdornment": <Unknown
                position="start"
            >
                <Unknown
                    size="2.5rem"
                />
            </Unknown>,
            }
          }
          onChange={[Function]}
          placeholder="Cost"
          value=""
        />
      </div>
      <div
        className="input-group__date"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <[object Object]
      aria-label="add"
      color="primary"
      type="submit"
      variant="extended"
    >
      Save
    </[object Object]>
  </form>
</div>
`;

exports[`should render error for invalid form submission 2`] = `
<div>
  <p>
    Please provide description and amount
  </p>
  <form
    onSubmit={[Function]}
  >
    <CustomInput
      formControlProps={
        Object {
          "fullWidth": true,
        }
      }
      id="float"
      inputProps={
        Object {
          "placeholder": "Description",
          "startAdornment": <Unknown
            position="start"
        >
            <Unknown
                size="2.5rem"
            />
        </Unknown>,
        }
      }
      onChange={[Function]}
      value=""
    />
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <CustomInput
          formControlProps={
            Object {
              "fullWidth": true,
            }
          }
          id="float"
          inputProps={
            Object {
              "placeholder": "Cost",
              "startAdornment": <Unknown
                position="start"
            >
                <Unknown
                    size="2.5rem"
                />
            </Unknown>,
            }
          }
          onChange={[Function]}
          placeholder="Cost"
          value=""
        />
      </div>
      <div
        className="input-group__date"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <[object Object]
      aria-label="add"
      color="primary"
      type="submit"
      variant="extended"
    >
      Save
    </[object Object]>
  </form>
</div>
`;

exports[`should return expense form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <CustomInput
      formControlProps={
        Object {
          "fullWidth": true,
        }
      }
      id="float"
      inputProps={
        Object {
          "placeholder": "Description",
          "startAdornment": <Unknown
            position="start"
        >
            <Unknown
                size="2.5rem"
            />
        </Unknown>,
        }
      }
      onChange={[Function]}
      value=""
    />
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <CustomInput
          formControlProps={
            Object {
              "fullWidth": true,
            }
          }
          id="float"
          inputProps={
            Object {
              "placeholder": "Cost",
              "startAdornment": <Unknown
                position="start"
            >
                <Unknown
                    size="2.5rem"
                />
            </Unknown>,
            }
          }
          onChange={[Function]}
          placeholder="Cost"
          value=""
        />
      </div>
      <div
        className="input-group__date"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <[object Object]
      aria-label="add"
      color="primary"
      type="submit"
      variant="extended"
    >
      Save
    </[object Object]>
  </form>
</div>
`;

exports[`should return expense form with expese data 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <CustomInput
      formControlProps={
        Object {
          "fullWidth": true,
        }
      }
      id="float"
      inputProps={
        Object {
          "placeholder": "Description",
          "startAdornment": <Unknown
            position="start"
        >
            <Unknown
                size="2.5rem"
            />
        </Unknown>,
        }
      }
      onChange={[Function]}
      value="item 2"
    />
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <CustomInput
          formControlProps={
            Object {
              "fullWidth": true,
            }
          }
          id="float"
          inputProps={
            Object {
              "placeholder": "Cost",
              "startAdornment": <Unknown
                position="start"
            >
                <Unknown
                    size="2.5rem"
                />
            </Unknown>,
            }
          }
          onChange={[Function]}
          placeholder="Cost"
          value="50670"
        />
      </div>
      <div
        className="input-group__date"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1969-12-28T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <[object Object]
      aria-label="add"
      color="primary"
      type="submit"
      variant="extended"
    >
      Save
    </[object Object]>
  </form>
</div>
`;

exports[`should set amount if valid 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__itemLong"
      >
        <input
          autoFocus={true}
          className="input__long"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Amount"
          type="text"
          value="23.50"
        />
      </div>
      <div
        className="input-group__item"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <textarea
      className="textarea"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button
      className="button-sytle__edit"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`should set description on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__itemLong"
      >
        <input
          autoFocus={true}
          className="input__long"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value="New Description"
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Amount"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <textarea
      className="textarea"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value=""
    />
    <button
      className="button-sytle__edit"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`should set note on text change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__itemLong"
      >
        <input
          autoFocus={true}
          className="input__long"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Amount"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
            Object {
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
    <textarea
      className="textarea"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      value="some notes"
    />
    <button
      className="button-sytle__edit"
    >
      Save
    </button>
  </form>
</div>
`;
